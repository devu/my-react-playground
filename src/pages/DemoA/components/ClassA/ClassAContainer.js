import React, { Component } from 'react'

import { Grid, Segment, Label } from 'semantic-ui-react'
import ClassA from './ClassA'
import ClassB from '../ClassB/ClassB'

class ClassAContainer extends Component {
  constructor() {
    super()
    this.state = {
      test: 'test'
    }
  }

  async componentDidMount() {
    console.log('ClassDContainer::componentDidMount')
  }

  render() {
    return (
      <Segment>
        <Label pointing="below">ClassA Container</Label>
        <Grid columns={2} stackable>
          <Grid.Column>
            <ClassA name="Class A" />
          </Grid.Column>
          <Grid.Column>
            <ClassB name="Class B" />
          </Grid.Column>
        </Grid>
      </Segment>
    )
  }
}

export default ClassAContainer
