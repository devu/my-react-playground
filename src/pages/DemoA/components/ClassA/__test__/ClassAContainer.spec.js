import React from 'react'
import { shallow } from 'enzyme'

import ClassAContainer from '../ClassAContainer'

/*
UNIT TEST SHOULD:
Isolated: all interactions with external services should are mocked
Specific: if change small functionality you would like to get specific test failure message
Descriptive: describe what the system does not how so that you can easily refactor
*/

const setup = propOverrides => {
  const props = Object.assign({
    test: 'test',
    onClear: jest.fn(),
  }, propOverrides)

  const wrapper = shallow(<ClassAContainer {...props} />)

  return {
    props,
    wrapper
    // clear: wrapper.find('.clear-completed'),
    // count: wrapper.find('.todo-count'),
  }
}

describe('ClassAContainer', () => {
  const { wrapper } = setup()

  it('should render', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
