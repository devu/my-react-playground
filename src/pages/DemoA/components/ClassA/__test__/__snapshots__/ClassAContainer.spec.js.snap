// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassAContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClassAContainer
    onClear={[MockFunction]}
    test="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Label
          pointing="below"
        >
          ClassA Container
        </Label>,
        <Grid
          columns={2}
          stackable={true}
        >
          <GridColumn>
            <ClassA
              name="Class A"
            />
          </GridColumn>
          <GridColumn>
            <ClassB
              name="Class B"
            />
          </GridColumn>
        </Grid>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "ClassA Container",
          "pointing": "below",
        },
        "ref": null,
        "rendered": "ClassA Container",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridColumn>
              <ClassA
                name="Class A"
              />
            </GridColumn>,
            <GridColumn>
              <ClassB
                name="Class B"
              />
            </GridColumn>,
          ],
          "columns": 2,
          "stackable": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ClassA
                name="Class A"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "Class A",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ClassB
                name="Class B"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "Class B",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Label
            pointing="below"
          >
            ClassA Container
          </Label>,
          <Grid
            columns={2}
            stackable={true}
          >
            <GridColumn>
              <ClassA
                name="Class A"
              />
            </GridColumn>
            <GridColumn>
              <ClassB
                name="Class B"
              />
            </GridColumn>
          </Grid>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "ClassA Container",
            "pointing": "below",
          },
          "ref": null,
          "rendered": "ClassA Container",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridColumn>
                <ClassA
                  name="Class A"
                />
              </GridColumn>,
              <GridColumn>
                <ClassB
                  name="Class B"
                />
              </GridColumn>,
            ],
            "columns": 2,
            "stackable": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ClassA
                  name="Class A"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "Class A",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ClassB
                  name="Class B"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "Class B",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
